### DISCUSSION TITLE
想在ide右侧插槽中增加浏览器实时预览和调试功能

### DISCUSSION THREAD

**Discussion by baxiamali** [opensumi/core#2345] [Q&A] [answered]
想在ide右侧插槽中增加浏览器实时预览和调试功能（类似淘宝开发者工具右侧预览，下方调试），想问下预览调试组件的实现思路和参考代码，非常感谢

**Comment by erha19** [comment, marked as answer]
@baxiamali 参考代码可以看一下这里的案例 https://github.com/opensumi/opensumi-module-samples/pull/19/commits/f9a15581e082c2f8b4307e4a2c57c735eae78b2e.

**Reply by baxiamali** [reply]
非常感谢解答，参考您的代码实现了类似的布局，我想问的是，有没有可以参考能实现网页预览功能的相关代码或者思路。
网页预览和调试功能，可以借助ide-webview来实现吗？还是需要参考vscode的webview进行功能开发呢？

**Reply by erha19** [reply]
调试功能一般会采用引入 Chrome Devtools frontend 页面的方式实现，网页预览功能的模拟可以找一下是否有社区封装好的组件或能力，这块一般与业务关联较强，没有官方实现。

**Reply by baxiamali** [reply]
感谢指导~社区是指opensumi社区还是vscode社区呀，opensumi社区是我们这个讨论区吗

**Reply by erha19** [reply]
指开源社区

**Reply by baxiamali** [reply]
明白啦 非常感谢
